selfies/utils/smiles_utils.py

```diff
@@ -442,7 +446,7 @@ def _derive_smiles_from_fragment(
     token = atom_to_smiles(curr_atom)
     derived.append(token)
     attribution_maps.append(AttributionMap(
-        len(derived) - 1 + attribution_index,
+        _strlen(derived) - 1 + attribution_index,
```

Comment:
In addition, I am a little hesitant to make this change because it joins the entire `derived` list together every call.  Would it be possible to implement this as some sort of running counter instead?